nml/actions/actionD.py

```diff
@@ -94,6 +94,9 @@ def pre_process(self):
             if not self.param.can_assign():
                 raise generic.ScriptError("Trying to assign a value to the read-only variable '{}'".format(self.param.name), self.param.pos)
         elif isinstance(self.param, expression.Identifier):
+            if global_constants.identifier_refcount[self.param.value] == 0:
+                generic.print_warning("Named parameter '{}' is not referenced, ignoring.".format(self.param.value), self.param.pos)
+                return
```

Comment:
I think it would be better to have the warning where it's actually handled, before `return []`.

That makes it easier to see what the code-path is doing, and avoids splitting the logic into two completely different places.

(I see the similar messages, but they're fatal ones that exit immediately with no handling elsewhere).